### PR TITLE
UI-7944 - Refactor folder structure to have one folder per migration

### PR DESCRIPTION
**Pull request by eliasinho**
## Description

We shouldn't have files related to migrating from 4.3 to 5.0 mixed with files related to migrating 5.0 to 5.1.
This PR introduces a new folder structure: we have one folder per migration at the root of `src`. Basically, one folder whose name is `fromVersion_to_toVersion` contains files related to the from `fromVersion` to `toVersion`.

At this point, this PR simply:
- adds a `4.3_to_5.0` folder under `src`
- renames the current `migrateUIFolder` to `migrateUIFolder43To50`

## Testing procedure

The build and tests should pass.

### CODE REVIEW

**Comment by RaspberryChaos** [comment]
I think we should also separate the `__test_resources__` because different test resources will be needed for different versions because the types have changed. This would avoid confusion if `legacyUIFolder` is a 4.3 or a 5.0 ui folder for example.

We could either :
1. Move the current `__test_resources__` inside the `4.3_to_5.0` folder and create a separate one inside `5.0_to_5.1`.
2. Keep it where it is and add sub folders `4.3_to_5.0` and `5.0_to_5.1` inside the `__test_resources__` folder.

What do you think @eliasinho ?

**Comment by eliasinho** [comment]
> I think we should also separate the `__test_resources__` because different test resources will be needed for different versions because the types have changed. This would avoid confusion if `legacyUIFolder` is a 4.3 or a 5.0 ui folder for example.
> 
> We could either :
> 
> 1. Move the current `__test_resources__` inside the `4.3_to_5.0` folder and create a separate one inside `5.0_to_5.1`.
> 2. Keep it where it is and add sub folders `4.3_to_5.0` and `5.0_to_5.1` inside the `__test_resources__` folder.
> 
> What do you think @eliasinho ?

Ah yes very good point! The current test resources indeed seem to be specific to the migration from 4.3, so it makes sense to separate them from future test resources. I think it's better to have them inside the corresponding `fromVersion_to_toVersion` folder, if they're only used there.

I'll add a commit to move them, and assign @antoinetissier to the PR to have his opinion.

**Review by antoinetissier — COMMENTED** [review]
A few remarks but it looks good !
If we're going to refactor anyway, I think I like `migrate_43_to_50` better than `migrateUIFolder43To50`: it's easier to read the numbers at first glance imo, and it's better to remove "ui folder" because:
- it's not just the ui folders in some migrations (e.g. calculated measures from 5.0 to 5.1)
- in any case, it doesn't add too much value to add it in the name, when the migration CLI itself is just `yarn migrate`

**Comment by antoinetissier** [comment]
> We could either :
1. Move the current __test_resources__ inside the 4.3_to_5.0 folder and create a separate one inside 5.0_to_5.1.
2. Keep it where it is and add sub folders 4.3_to_5.0 and 5.0_to_5.1 inside the __test_resources__ folder.

I'd definitely go for 1: better to colocate resources with the code they are meant to help testing.

File: src/4.3_to_5.0/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export { migrateDashboard } from "./migrateDashboard";
+export { migrateWidget } from "./migrateWidget";
+export { migrateChart } from "./migrateChart";
+export { migrateTable } from "./migrateTable";
+export { migrateKpi } from "./migrateKpi";
+export { migrateQuickFilter } from "./migrateQuickFilter";
```
[the remark this answers is not in the document]
Comment (eliasinho) [reply src/4.3_to_5.0/index.ts:6]: Should we still export the types from `migration.types`?

File: src/index.ts [hunk 2]
```diff
@@ -1,9 +1 @@
-export { migrateDashboard } from "./migrateDashboard";
-export { migrateWidget } from "./migrateWidget";
-export { migrateChart } from "./migrateChart";
-export { migrateTable } from "./migrateTable";
-export { migrateKpi } from "./migrateKpi";
-export { migrateQuickFilter } from "./migrateQuickFilter";
-export { migrateUIFolder } from "./migrateUIFolder";
-
-export * from "./migration.types";
+export * from "./4.3_to_5.0";
```
[the remark this answers is not in the document]
Comment (eliasinho) [reply src/index.ts:1]: Related to my question [here](https://github.com/activeviam/activeui-migration/pull/54#discussion_r1029478573), but that means that we don't care about exporting `migration.types` at all, right?

**Review by RaspberryChaos — APPROVED** [review]
Looks good to me! 🚀

File: src/4.3_to_5.0/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoinetissier) [reply src/4.3_to_5.0/index.ts]: I think we should do the same for the types: let's not export them for now and we can always add them later

**Review by antoinetissier — APPROVED** [review]
🚀

**Pull request merged**